hugr/src/hugr/validate.rs

```diff
@@ -47,19 +44,68 @@ impl Hugr {
     /// variables.
     /// TODO: Add a version of validation which allows for open extension
     /// variables (see github issue #457)
+    #[cfg(feature = "extension_inference")]
     pub fn validate(&self, extension_registry: &ExtensionRegistry) -> Result<(), ValidationError> {
         self.validate_with_extension_closure(HashMap::new(), extension_registry)
     }
 
+    /// Check the validity of the HUGR, disregarding extension requirements.
+    #[cfg(not(feature = "extension_inference"))]
```

Comment:
Nice! :)

Maybe even common-up the signature and put the `#[cfg]` just inside the method body?